src/cpp/CMakeLists.txt

```
@@ -63,8 +63,7 @@ configure_file("${OpenVINOGenAI_SOURCE_DIR}/cmake/templates/version.cpp.in"
                "${CMAKE_CURRENT_BINARY_DIR}/version.cpp" @ONLY)
 
 # Object library
-
-file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c")
+file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c" "${CMAKE_SOURCE_DIR}/src/c/src/*.cpp")
```

```suggestion
file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c" "${OpenVINOGenAI_SOURCE_DIR}/src/c/src/*.cpp")
```
you need to use `OpenVINOGenAI_SOURCE_DIR` instead of `CMAKE_SOURCE_DIR` in all other places as well, because in case of OpenVINO GenAI is used as a submodule, `CMAKE_SOURCE_DIR` will be a different path and such lines will not work.

---

I've update the CMakeLists.txt and used `OpenVINOGenAI_SOURCE_DIR` here.